web_programming/fetch_anime_and_play.py

```diff
@@ -0,0 +1,205 @@
+from fake_useragent import UserAgent
+from requests.exceptions import RequestException
+
+BASE_URL = "https://ww1.gogoanime2.org"
+
+
+def search_scraper(anime_name: str) -> list:
+
+    """[summary]
+
+    This function will take an url and
+    return list of anime after scraping the site.
+
+    >>> type(search_scraper("demon_slayer"))
+    <class 'list'>
+
+    Args:
+        anime_name (str): [Name of anime]
+
+    Raises:
+        e: [Raises exception on failure]
+
+    Returns:
+        [list]: [List of animes]
+    """
+
+    # concat the name to form the search url.
+    search_url = f"{BASE_URL}/search/{anime_name}"
+
+    try:
+        response = requests.get(
+            search_url, headers={"UserAgent": UserAgent().chrome}
+        )  # request the url.
+
+        # Is the response ok?
+        response.raise_for_status()
+    except RequestException as e:
+        raise e
+
+    # parse with soup.
+    soup = BeautifulSoup(response.text, "html.parser")
+
+    # get list of anime
+    items_ul = soup.find("ul", {"class": "items"})
+    items_li = items_ul.children
+
+    # for each anime, insert to list. the name and url.
+    anime_list = []
+    for li in items_li:
+        if not isinstance(li, NavigableString):
+            try:
+                anime_url, anime_title = li.find("a")["href"], li.find("a")["title"]
+                anime_list.append(
+                    {
+                        "title": anime_title,
+                        "url": anime_url,
+                    }
+                )
+            except (NotFoundErr, KeyError):
+                pass
+
+    return anime_list
+
+
+def search_anime_episode_list(episode_endpoint: str) -> list:
+
+    """[summary]
+
+    This function will take an url and
+    return list of episodes after scraping the site
+    for an url.
+
+    >>> type(search_anime_episode_list("/anime/kimetsu-no-yaiba"))
+    <class 'list'>
+
+    Args:
+        episode_endpoint (str): [Endpoint of episode]
+
+    Raises:
+        e: [description]
+
+    Returns:
+        [list]: [List of episodes]
+    """
+
+    request_url = f"{BASE_URL}{episode_endpoint}"
+
+    try:
+        response = requests.get(
+            url=request_url, headers={"UserAgent": UserAgent().chrome}
+        )
+        response.raise_for_status()
+    except RequestException as e:
+        raise e
+
+    soup = BeautifulSoup(response.text, "html.parser")
+
+    # With this id. get the episode list.
+    episode_page_ul = soup.find("ul", {"id": "episode_related"})
+    episode_page_li = episode_page_ul.children
+
+    episode_list = []
+    for children in episode_page_li:
+        try:
+            if not isinstance(children, NavigableString):
+                episode_list.append(
+                    {
+                        "title": children.find("div", {"class": "name"}).text.replace(
+                            " ", ""
+                        ),
+                        "url": children.find("a")["href"],
```

Comment:
It confuses the reader that children is plural yet these items are singular.  It is easier to understand when each item is an episode...
```suggestion
    for episode in episode_page_li:
        try:
            if not isinstance(episode, NavigableString):
                episode_list.append(
                    {
                        "title": episode.find("div", {"class": "name"}).text.replace(
                            " ", ""
                        ),
                        "url": episode.find("a")["href"],
```

Reply:
for episode in episode_page_li:
for anime in anime_li:
Like this now